android/quest/src/test/java/org/smartregister/fhircore/quest/ui/login/LoginViewModelTest.kt

```diff
@@ -617,8 +620,48 @@ internal class LoginViewModelTest : RobolectricTest() {
 
   @Test
   fun testForgotPasswordLoadsContact() {
-    loginViewModel.forgotPassword()
-    Assert.assertEquals("tel:0123456789", loginViewModel.launchDialPad.value)
+    val context = ApplicationProvider.getApplicationContext<Context>()
+    val validContactNumber = "1234567890"
+    val expectedFormattedNumber = "+1-123-456-7890"
+
+    every { loginViewModel.applicationConfiguration.loginConfig.supervisorContactNumber } returns
+      validContactNumber
+    every { loginViewModel.formatPhoneNumber(context, validContactNumber) } returns
+      expectedFormattedNumber
+
+    val dialPadUriSlot = slot<String?>()
+    val launchDialPadObserver =
+      Observer<String?> { dialPadUri -> dialPadUriSlot.captured = dialPadUri }
+
+    loginViewModel.launchDialPad.observeForever(launchDialPadObserver)
+
+    try {
+      loginViewModel.forgotPassword(context)
+      Assert.assertEquals(expectedFormattedNumber, dialPadUriSlot.captured)
+    } finally {
+      loginViewModel.launchDialPad.removeObserver(launchDialPadObserver)
+    }
+  }
+
+  @Test
+  fun testForgotPasswordWithValidContactNumber() {
+    val context = ApplicationProvider.getApplicationContext<Context>()
+    val validContactNumber = "1234567890"
+    val expectedFormattedNumber = "+1-123-456-7890"
+
+    every { loginViewModel.applicationConfiguration.loginConfig.supervisorContactNumber } returns
+      validContactNumber
+    every { loginViewModel.formatPhoneNumber(context, validContactNumber) } returns
+      expectedFormattedNumber
```

Comment:
It is not necessary to mock this function. Can you update this test without mocking the function?